### ISSUE TITLE
File reported as not matching project config when using JSX

### ISSUE THREAD

**Issue by kirkobyte** [typescript-eslint/typescript-eslint#2850]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

The `tsconfig.json` includes `jsx: preserve`. The error still occurs whether `includes` is configured in a way that files will work if they don't have a `.tsx` extension.

```TS
// Test file
const ruleTester = new ESLintUtils.RuleTester({
  parser: '@typescript-eslint/parser',
  parserOptions: {
    extraFileExtensions: ['.tsx'], // A warning is provided stating that this is unnecessary. But the file is still reported as not matching the project config
    ecmaVersion: 2018,
    ecmaFeatures: {
      jsx: true,
    },
    sourceType: 'module',
    project: './tsconfig.json',
    // project: './tests/tsconffig.test.json', // relative to tsconfigRootDir
  },
});

ruleTester.run('deprecation', rule, {
  valid: [
    {
      code: `
        const component = <Component/>;
      `,
      filename: 'file.tsx',
    },
  ],
  // Error cases. `// ERROR: x` marks the spot where the error occurs.
  invalid: [],
});
```

**Expected Result**

The provided test should pass.

**Actual Result**

The file is reported as not matching one of the projects provided, even though everything is configured to enable JSX and the files are in an appropriate location where  `.ts` files work.

<img width="1215" alt="Screen Shot 2020-12-04 at 1 33 08 PM" src="https://user-images.githubusercontent.com/6798667/101164407-485aaa00-3635-11eb-8e90-16d389e68a9d.png">

**Additional Info**

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `4.8.1` |
| `TypeScript`                            | `4.1.2` |
| `node`                                  | `12.18.3` |

**Comment by kirkobyte** [comment]
@gund and I found this issue while attempting to add JSX support to the `eslint-plugin-deprecation` project and its tests https://github.com/gund/eslint-plugin-deprecation/pull/15

**Comment by bradzacher** [comment]
You do not need `extraFileExtensions` for tsx files.
That option is required for non standard extensions only, like `vue`

**Comment by kirkobyte** [comment]
It didn't work without `extraFileExtensions`. But @caspardue pointed out that we needed to add a mock file with the `.tsx` extension. It'd be nice to not require the mock files, but we've sorted this for now. Thanks!

**Comment by bradzacher** [comment]
TypeScript is built on top of  the file system, which is why a dummy file is required on disk along with a tsconfig referencing it.

Its possible to switch testing across to a virtual file system, but that requires someone to take time and effort to investigate and build.

https://github.com/typescript-eslint/typescript-eslint/issues/1891

🤷 The community would rather use dummy files instead of spending time investigating.

**Comment by kirkobyte** [comment]
That's fair! Thanks for the explanation 😄
